### PR TITLE
[MAINTENANCE] Placing S3, Trino Imports Under Compatibility Pattern

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Instead of having plain imports of `boto3`/`botocore` and `trino`, they are placed under the established "Compatibility" coding pattern.

### Remarks
* JIRA: DX-1/DX-440

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/compatibility/aws.py [hunk 1]
```diff
@@ -3,21 +3,20 @@
...
 
 try:
-    import boto3
+    import boto3  # noqa TID251
 except ImportError:
     boto3 = AWS_NOT_IMPORTED
 
 try:
-    import botocore
+    import botocore  # TID251
```
Comment (Shinnnyshinshin) [inline great_expectations/compatibility/aws.py]: I'm seeing this one as ['banned API`](https://beta.ruff.rs/docs/rules/banned-api/). Is there a reason why we weren't seeing this before?

File: pyproject.toml [hunk 2]
```diff
@@ -345,8 +345,10 @@ convention = "google"
 [tool.ruff.flake8-tidy-imports.banned-api]
 "sqlalchemy".msg = "Please do not import sqlalchemy directly, import from great_expectations.compatibility.sqlalchemy instead."
 "pyspark".msg = "Please do not import pyspark directly, import from great_expectations.compatibility.pyspark instead."
+"boto3".msg = "Please do not import boto3 directly, import from great_expectations.compatibility.aws instead."
```
Comment (Shinnnyshinshin) [inline pyproject.toml:348]: 🚀

File: pyproject.toml [hunk 3]
```diff
@@ -345,8 +345,10 @@ convention = "google"
...
 "google".msg = "Please do not import google directly, import from great_expectations.compatibility.google instead."
 "azure".msg = "Please do not import azure directly, import from great_expectations.compatibility.azure instead."
+"trino".msg = "Please do not import trino directly, import from great_expectations.compatibility.trino instead."
```
Comment (Shinnnyshinshin) [inline pyproject.toml:351]: 🚀

File: tests/expectations/metrics/test_core.py [hunk 4]
```diff
@@ -77,6 +77,98 @@ def test_basic_metric_pd():
...
+            pd.DataFrame({"a": [1, 2, 3, None]}),
+            6,
+        ],
+        [
+            pd.DataFrame({"a": [Decimal(2.0), Decimal(0.18781)]}),
+            2.18781,
+        ],
+    ],
+)
+def test_column_sum_metric_spark(spark_session, dataframe, expected_result):
```
Comment (Shinnnyshinshin) [inline tests/expectations/metrics/test_core.py]: This feels out of scope for placing S3 and Trino under the `Compatiblity` pattern. Is there a reason why this test is being included?

File: great_expectations/compatibility/aws.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/compatibility/aws.py]: @Shinnnyshinshin I am still trying to figure out how to avoid this `# TID251`.  Thanks.

File: tests/expectations/metrics/test_core.py (hunk 4 above)
Comment (alexsherstinsky) [reply tests/expectations/metrics/test_core.py]: This will be merged from another PR (approved).  Thanks.

File: great_expectations/compatibility/aws.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/compatibility/aws.py]: Hmmm interesting.  Maybe @Kilo59 could weigh in on this... it just feels a bit wrong to override this

File: tests/expectations/metrics/test_core.py (hunk 4 above)
Comment (Shinnnyshinshin) [reply tests/expectations/metrics/test_core.py]: 👍🏼

File: great_expectations/compatibility/aws.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/compatibility/aws.py]: @alexsherstinsky it looks like there are other places in the codebase that have this flag, including a few that were introduced by this PR. `https://github.com/great-expectations/great_expectations/pull/8123/` for sqlalchemy. I think we are ok to go here for now

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM
